sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/SearchableField.java

```diff
@@ -0,0 +1,146 @@
+     * changed for the field in the index.
+     *
+     * @return The {@link AnalyzerName} used for search analyzer.
+     */
+    public AnalyzerName getSearchAnalyzer() {
+        return searchAnalyzer;
+    }
+
+    /**
+     * Sets the name of the language analyzer for searching. This property must be set together with
+     * {@code indexAnalyzer}, and cannot be set when {@code analyzer} is set. Once the analyzer is chosen, it cannot be
+     * changed for the field in the index.
+     *
+     * @param searchAnalyzer The {@link AnalyzerName} used for search analyzer.
+     * @return The SearchableField object itself.
+     */
+    public SearchableField setSearchAnalyzer(AnalyzerName searchAnalyzer) {
+        this.searchAnalyzer = searchAnalyzer;
+        return this;
+    }
+
+    /**
+     * Gets the name of the language analyzer for indexing. This property must be set together with
+     * {@code searchAnalyzer}, and cannot be set when {@code analyzer} is set. Once the analyzer is chosen, it cannot be
+     * changed for the field in the index.
+     *
+     * @return The {@link AnalyzerName} used for index analyzer.
+     */
+    public AnalyzerName getIndexAnalyzer() {
+        return indexAnalyzer;
+    }
+
+    /**
+     * Gets the name of the language analyzer for indexing. This property must be set together with
+     * {@code searchAnalyzer}, and cannot be set when {@code analyzer} is set. Once the analyzer is chosen, it cannot be
+     * changed for the field in the index.
+     *
+     * @param indexAnalyzer The {@link AnalyzerName} used for index analyzer.
+     * @return The SearchableField object itself.
+     */
+    public SearchableField setIndexAnalyzer(AnalyzerName indexAnalyzer) {
+        this.indexAnalyzer = indexAnalyzer;
+        return this;
+    }
+
+    /**
+     * Gets a list of names of synonym maps to associate with this field.
+     * Currently, only one synonym map per field is supported.
+     *
+     * Assigning a synonym map to a field ensures that query terms targeting that field are expanded at query-time using
+     * the rules in the synonym map. This attribute can be changed on existing fields.
+     *
+     * @return List of names of synonym maps to associate with this field.
+     */
+    public List<String> getSynonymMaps() {
+        return synonymMaps;
+    }
+
+    /**
+     * Sets a list of names of synonym maps to associate with this field.
+     * Currently, only one synonym map per field is supported.
+     *
+     * Assigning a synonym map to a field ensures that query terms targeting that field are expanded at query-time using
+     * the rules in the synonym map. This attribute can be changed on existing fields.
+     *
+     * @param synonymMaps list of names of synonym maps to associate with this field.
+     * @return The SearchableField object itself.
+     */
+    public SearchableField setSynonymMaps(List<String> synonymMaps) {
```

Comment:
I'd call this `synonymMapNames` since it's the names of synonymMaps. IIRC, that's what I did for that reason.